xapian-bindings/python3/Makefile.am

```diff
@@ -172,3 +174,110 @@ $(sphinxdocs): xapian/__init__.py xapian/_xapian$(PYTHON3_SO) docs/conf.py $(RST
+
+EXTRA_DIST += pythonletor.i \
+	exceptletor.i \
+	generate-letor-python-exceptions \
+	test_xapian_letor_star.py \
+	pythonletortest.py \
+	xapianletor.py \
+	xapianletor_wrap.cc \
+	xapianletor_wrap.h
+
+letorpkgpylibdir = @PYTHON3_LIB@/xapianletor
+
+# Install as _DATA rather than _SCRIPTS because we don't want to make these
+# executable (they don't have a #! line).
+letorpkgpylib_DATA = xapianletor/__init__.py \
+		xapianletor/__pycache__/__init__.@PYTHON3_CACHE_TAG@.pyc \
+		xapianletor/__pycache__/__init__.@PYTHON3_CACHE_OPT1_EXT@
+
+letorpkgpylib_LTLIBRARIES = _xapianletor.la
+
+_xapianletor_la_CXXFLAGS = $(SWIG_CXXFLAGS) $(XAPIAN_CXXFLAGS) $(XAPIANLETOR_CXXFLAGS)
+_xapianletor_la_LDFLAGS = -avoid-version -module -shrext "$(PYTHON3_SO)" $(NO_UNDEFINED)
+_xapianletor_la_SOURCES = xapianletor_wrap.cc
+_xapianletor_la_LIBADD = $(XAPIANLETOR_LIBS) $(PYTHON3_LIBS)
+
+# We want `xapianletor/__init__.py` rather than xapianletor.py so the module is treated
+# as a package.
+xapianletor/__init__.py: xapian/__init__.py xapianletor.py
+	$(MKDIR_P) xapianletor
+	cp `test -f xapianletor.py || echo '$(srcdir)/'`xapianletor.py xapianletor/__init__.py
+
+# We "import _xapianletor" first so that if we fail to import the glue library
+# we don't generate a broken .pyc or .pyo, and we do it with the current
+# directory set to where the module is as Python with -c always searches the
+# current directory first.
+#
+# We then "import xapianletor" as a separate command, as that's more like how things
+# are once installed - in particular this will fail if the SWIG shadow wrapper
+# doesn't cope with being xapianletor/__init__.py (currently it needs fixing
+# up, which is done by fixup-swig-py3-wrapper).
+xapianletor/__pycache__/__init__.@PYTHON3_CACHE_TAG@.pyc: xapian/__pycache__/__init__.@PYTHON3_CACHE_TAG@.pyc xapianletor/__init__.py xapianletor/_xapianletor$(PYTHON3_SO)
+	$(OSX_SIP_HACK_ENV) $(PYTHON3) -c 'import os;os.chdir("xapianletor");import _xapianletor'
+	$(OSX_SIP_HACK_ENV) $(PYTHON3) -c 'import xapianletor'
+
+xapianletor/__pycache__/__init__.@PYTHON3_CACHE_OPT1_EXT@: xapian/__pycache__/__init__.@PYTHON3_CACHE_OPT1_EXT@ xapianletor/__init__.py xapianletor/_xapianletor$(PYTHON3_SO)
+	$(OSX_SIP_HACK_ENV) $(PYTHON3) -O -c 'import os;os.chdir("xapianletor");import _xapianletor'
+	$(OSX_SIP_HACK_ENV) $(PYTHON3) -O -c 'import xapianletor'
+
+xapianletor/_xapianletor$(PYTHON3_SO): _xapian.la _xapianletor.la
+	$(MKDIR_P) xapianletor
+	$(LIBTOOL) --config > libtoolconfig.tmp
+## ksh requires a path on the sourced file.
+	. ./libtoolconfig.tmp; cp $$objdir/_xapianletor$(PYTHON3_SO) xapianletor
+	rm -f libtoolconfig.tmp
+
+CLEANFILES += \
+    xapianletor/_xapianletor$(PYTHON3_SO) \
+    xapianletor/__init__.py \
+    xapianletor/__pycache__/__init__.@PYTHON3_CACHE_TAG@.pyc \
+    xapianletor/__pycache__/__init__.@PYTHON3_CACHE_OPT1_EXT@ \
+    err_output_ListNet_2.txt \
+    ndcg_output_ListNet_2.txt \
+    training_output1.txt \
+    $(srcdir)/*.pyc
```

Comment:
`$(srcdir)/*.pyc` is already in `CLEANFILES` from above.

Reply:
fixed in a fixup commit.